src/Microsoft.SymbolStore/SymbolStores/HttpSymbolStore.cs

```diff
@@ -119,7 +119,7 @@ protected override async Task<SymbolStoreFile> GetFileInner(SymbolStoreKey key,
 
         protected Uri GetRequestUri(string index)
         {
-            if (!Uri.TryCreate(Uri, Uri.EscapeDataString(index), out Uri requestUri))
```

Comment:
I moved the index escaping from the key generations in KeyGenerator.cs to here about a year ago in PR #170. I don't think the fix is as simple as removing the escaping here.  Need to do more investigation on why this is happening now.

Reply:
True, it was working with escaping until last week of January at least (last time I tried on a fresh instance). I think this is due to some change on the storage side. It also redirects from http to https now (which happens seamlessly).

Reply:
In the end, we still need to escape spaces and other invalid characters.  We (all the teams at MS that use the uploader and downloader based on symstore) do have files that have spaces in them that we need to download.  

I'm going to suggest that we use Uri.EscapeUriString here.  There is some controversy (see the last paragraph [here](https://stackoverflow.com/questions/4396598/whats-the-difference-between-escapeuristring-and-escapedatastring)) about this API vs Uri.EscapeDataString. EscapeUriString escapes only illegal/invalid characters like space where EscapeDataString also escapes "reserved" chars like "/".  EscapeUriString has the downside that it can double escape already escaped characters but that won't happen here because "index" is never already "escaped". 

/cc: @hoyosjs

Reply:
Makes sense. Using `EscapeUriString` for `index` also worked:

Original: `_.debug/elf-buildid-sym-27997343859af70e0bfc5e50a4d1059b3513a1c4/_.debug`
EscapeUriString : `_.debug/elf-buildid-sym-27997343859af70e0bfc5e50a4d1059b3513a1c4/_.debug`
EscapeDataString: `_.debug%2Felf-buildid-sym-27997343859af70e0bfc5e50a4d1059b3513a1c4%2F_.debug`